fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/postgresql/PostgreSqlResourceNoProcDAO.java

```diff
@@ -0,0 +1,541 @@
+
+            resource.setId(resourceId);
+            if (logger.isLoggable(Level.FINE)) {
+                logger.fine("Successfully inserted Resource. id=" + resource.getId() + " executionTime=" + dbCallDuration + "ms");
+            }
+        } catch(FHIRPersistenceDBConnectException | FHIRPersistenceDataAccessException e) {
+            throw e;
+        } catch(SQLIntegrityConstraintViolationException e) {
+            FHIRPersistenceFKVException fx = new FHIRPersistenceFKVException("Encountered FK violation while inserting Resource.");
+            throw severe(logger, fx, e);
+        } catch(SQLException e) {
+            if ("99001".equals(e.getSQLState())) {
+                // this is just a concurrency update, so there's no need to log the SQLException here
+                throw new FHIRPersistenceVersionIdMismatchException("Encountered version id mismatch while inserting Resource");
+            } else {
+                FHIRPersistenceException fx = new FHIRPersistenceException("SQLException encountered while inserting Resource.");
+                throw severe(logger, fx, e);
+            }
+        } catch(Throwable e) {
+            FHIRPersistenceDataAccessException fx = new FHIRPersistenceDataAccessException("Failure inserting Resource.");
+            throw severe(logger, fx, e);
+        } finally {
+            logger.exiting(CLASSNAME, METHODNAME);
+        }
+
+        return resource;
+
+    }
+    
+    /**
+     * Store the resource in the database, creating a new logical_resource entry if this is
+     * the first version of this resource, or creating a new resource entry if this a new
+     * version of an existing logical resource. The logic tracks closely the DB2 stored
+     * procedure implementation, including locking of the logical_resource and handling
+     * concurrency issues using the standard insert-or-update pattern:
+     * <pre>
+     *   SELECT FOR UPDATE                 -- try and get a write lock
+     *   IF NOT FOUND THEN                 -- doesn't exist, so we don't have a lock
+     *     INSERT new logical resource     -- create the record - if OK, we own the lock
+     *     IF DUPLICATE THEN               -- someone else beat us to the create
+     *       SELECT FOR UPDATE             -- so we need to try again for a write lock
+     *     ...
+     *   ...
+     * </pre>
+     *
+     * This works because we never delete a logical_resource record, and so don't have to deal
+     * with concurrency issues caused when deletes are mingled with inserts/updates
+     *
+     * Note the execution flow aligns very closely with the DB2 stored procedure
+     * implementation (fhir-persistence-schema/src/main/resources/add_any_resource.sql)
+     *
+     * @param tablePrefix
+     * @param parameters
+     * @param p_logical_id
+     * @param p_payload
+     * @param p_last_updated
+     * @param p_is_deleted
+     * @param p_source_key
```

Comment:
Note, I have a PR which will be invalidated once this is in.  I'll update to take account of this. 

Just an FYI